src/bun.js/webcore.zig

```diff
@@ -557,27 +551,25 @@ pub const Crypto = struct {
             buf.ptr,
             keylen,
         ) != 1) {
-            return throwInvalidParams(globalThis, "RangeError", "Invalid scrypt params\n", .{});
+            return throwInvalidParams(globalThis, "Invalid scrypt params\n", .{});
         }
 
         return JSC.ArrayBuffer.create(globalThis, buf, .ArrayBuffer);
     }
 
     fn throwInvalidParameter(globalThis: *JSC.JSGlobalObject) JSC.JSValue {
-        const err = globalThis.createError(
+        const err = globalThis.createErrorInstanceWithCode(
             .ERR_CRYPTO_SCRYPT_INVALID_PARAMETER,
-            "Error",
             "Invalid scrypt parameters",
             .{},
         );
         globalThis.throwValue(err);
         return .zero;
     }
 
-    fn throwInvalidParams(globalThis: *JSC.JSGlobalObject, comptime error_name: string, comptime message: string, fmt: anytype) JSC.JSValue {
-        const err = globalThis.createError(
+    fn throwInvalidParams(globalThis: *JSC.JSGlobalObject, comptime message: string, fmt: anytype) JSC.JSValue {
+        const err = globalThis.createRangeErrorInstanceWithCode(
```

Comment:
Are all invalid params range errors?

Reply:
They can also be type errors

Reply:
Oh in scryptSync I'm pretty sure they are all range errors

Reply:
maybe keep it as an arg but use a `@Type(.EnumLiteral)`

then in here check which one and call the appropriate function

Reply:
👍